class/defaults.yml

```diff
@@ -45,8 +45,17 @@ parameters:
 
     nodeGroups: {}
 
-    containerRuntimeConfigs: {}
+    machineConfigPools:
+      # master and worker pools are not managed by the component
+      master: {}
+      worker: {}
```

Comment:
why are these fields here if we're never going to manage them anyway?

Reply:
Well while we can't manage the pool spec through the component we can manage the kubelet, containerruntime, and additional machine configs for these pools through this parameter.

But that probably wasn't clearly stated. I'll definitely invest some more time to properly document this.

Reply:
IMO that's only relevant if the goal is to drop the top-level `kubeletConfigs`, `containerRuntimeConfigs` and `machineConfigs` in the future (in which case I'd expect that users get a deprecation warning if any config is found in those parameters).